engine/src/main/java/org/terasology/world/generation/facets/base/BaseBooleanFieldFacet2D.java

```diff
@@ -31,8 +31,7 @@
 
     public BaseBooleanFieldFacet2D(BlockRegion targetRegion, Border3D border) {
         super(targetRegion, border);
-        Vector2i size = getRelativeRegion().size();
-        data = new boolean[size.x * size.y];
+        data = new boolean[getRelativeRegion().getSizeX() * getRelativeRegion().getSizeY()];
```

Comment:
I'm somewhat confused - don't we typically treat a 2D variant of some region as projection onto the XZ plane? Should this have been `size.x * size.z` before? 🤔

Reply:
you might of found a bug?

Reply:
nope getRelativeRegion() is a BlockArea so its x and y.

Reply:
alright, then this also boils down to renaming that method `getRelativeArea()` 😉